django/contrib/postgres/forms/jsonb.py

```diff
@@ -0,0 +1,29 @@
+import json
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+__all__ = ['JSONField']
+
+
+class JSONField(forms.CharField):
+    default_error_messages = {
+        'invalid': _("'%(value)s' value must be valid JSON."),
+    }
+
+    def __init__(self, **kwargs):
+        kwargs.setdefault('widget', forms.Textarea)
+        super(JSONField, self).__init__(**kwargs)
+
+    def to_python(self, value):
+        try:
+            return json.loads(value)
+        except ValueError:
```

Comment:
If this is a genuine security concern, then we shouldn't ship an actual `JSONField` at all, but then we make ourselves vulnerable to someone not checking user data at all and we get into the problem Markus has suggested.

Markus - the form layer would normally do this kind of validation I think, which is what's going on here. The model layer will receive the decoded JSON object to save to the database always. (If it got a string, this could just be a valid "JSON string" i.e. a string literal).